--- /usr/local/petget/check_deps.sh	2010-09-01 15:50:10.000000000 +0800
+++ /usr/local/petget/check_deps_table.sh	2010-11-29 21:17:26.000000000 +0800
@@ -6,6 +6,11 @@
 #100718 reduce size of missing-libs list, to fit in window.
 #100830 missing libs, but some pkgs have a startup script that makes some libs visible.
 
+Pro="/check_deps.sh"
+Script_Dir="/usr/local/petget"
+if test -f $Script_Dir/path ; then . $Script_Dir/path ; fi
+DBG="$Script_Dir$Pro line :"  ##krg---debugging value---<<<
+
 . /etc/DISTRO_SPECS #has DISTRO_BINARY_COMPAT, DISTRO_COMPAT_VERSION
 . /root/.packages/DISTRO_PKGS_SPECS
 
@@ -134,7 +139,7 @@
  ACTIONBUTTON="<button>
      <label>Check dependencies</label>
      <action type=\"exit\">BUTTON_CHK_DEPS</action>
-    </button>"
+     </button>"
  echo -n "" > /tmp/PetGet/petget_depchk_buttons
  cat /root/.packages/user-installed-packages | cut -f 1,10 -d '|' |
  while read ONEPKGSPEC
@@ -142,27 +147,33 @@
   [ "$ONEPKGSPEC" = "" ] && continue
   ONEPKG="`echo -n "$ONEPKGSPEC" | cut -f 1 -d '|'`"
   ONEDESCR="`echo -n "$ONEPKGSPEC" | cut -f 2 -d '|'`"
-  echo "<radiobutton><label>${ONEPKG} DESCRIPTION: ${ONEDESCR}</label><variable>RADIO_${ONEPKG}</variable></radiobutton>" >> /tmp/PetGet/petget_depchk_buttons
+  echo "${ONEPKG} DESCRIPTION: ${ONEDESCR}" >> /tmp/PetGet/petget_depchk_buttons
  done
  RADBUTTONS="`cat /tmp/PetGet/petget_depchk_buttons`"
- if [ "$RADBUTTONS" != "" ];then #100902 test if too many installed pkgs
-  nRADBUT=`echo "$RADBUTTONS" | wc -l`
-  if [ $nRADBUT -gt 23 ];then
-   #layout as two columns...
-   RADBUTTONS1="`cat /tmp/PetGet/petget_depchk_buttons | head -n 23`"
-   RADBUTTONS2="`cat /tmp/PetGet/petget_depchk_buttons | tail -n +24`"
-   RADBUTTONS="<hbox><vbox>${RADBUTTONS1}</vbox><vbox>${RADBUTTONS2}</vbox></hbox>"
-  fi
- fi
+ #if [ "$RADBUTTONS" != "" ];then #100902 test if too many installed pkgs
+ # nRADBUT=`echo "$RADBUTTONS" | wc -l`
+ # if [ $nRADBUT -gt 23 ];then
+ #  #layout as two columns...
+ #  RADBUTTONS1="`cat /tmp/PetGet/petget_depchk_buttons | head -n 23`"
+ #  RADBUTTONS2="`cat /tmp/PetGet/petget_depchk_buttons | tail -n +24`"
+ #  RADBUTTONS="<hbox><vbox>${RADBUTTONS1}</vbox><vbox>${RADBUTTONS2}</vbox></hbox>"
+ # fi
+ #fi
  if [ "$RADBUTTONS" = "" ];then
   ACTIONBUTTON=""
-  RADBUTTONS="<text use-markup=\"true\"><label>\"<b>No packages installed by user, click 'Cancel' button</b>\"</label></text>"
+  #RADBUTTONS="<text use-markup=\"true\"><label>\"<b>No packages installed by user, click 'Cancel' button</b>\"</label></text>"
+  echo "No packages installed by user, click 'Cancel' button." > /tmp/PetGet/petget_depchk_buttons
  fi
  export DEPS_DIALOG="<window title=\"Puppy Package Manager\" icon-name=\"gtk-about\">
   <vbox>
    <text><label>Please choose what package you would like to check the dependencies of:</label></text>
    <frame User-installed packages>
-   ${RADBUTTONS}
+    <table>
+     <label>Package List and Description           </label>
+      <variable>ONEPKG</variable>
+       <input>cat /tmp/PetGet/petget_depchk_buttons</input>
+       <width>700</width><height>180</height>
+    </table>
    </frame>
    <hbox>
     ${ACTIONBUTTON}
@@ -170,7 +181,7 @@
    </hbox>
   </vbox>
  </window>
-" 
+"
  RETPARAMS="`gtkdialog3 --program=DEPS_DIALOG`"
  #ex returned:
  #RADIO_audacious-1.5.1="true"
@@ -180,7 +191,11 @@
  #[ "$EXIT" != "BUTTON_CHK_DEPS" ] && exit
  [ "`echo "$RETPARAMS" | grep 'BUTTON_CHK_DEPS'`" = "" ] && exit
  
- APKGNAME="`echo "$RETPARAMS" | grep '^RADIO_' | grep '"true"' | cut -f 1 -d '=' | cut -f 2 -d '_'`"
+ # APKGNAME="`echo "$RETPARAMS" | grep '^RADIO_' | grep '"true"' | cut -f 1 -d '=' | cut -f 2 -d '_'`"
+ APKGNAME="`echo "$RETPARAMS" | grep ONEPKG | cut -f 2 -d '\"' | cut -f 1 -d ' '`"
+ echo $DBG 191 ONEPKG $ONEPKG
+ echo $DBG 192 RETPARAMS $RETPARAMS
+ echo $DBG 193 APKGNAME $APKGNAME
  dependcheckfunc
  
 fi
@@ -225,6 +240,6 @@
  </window>
 " 
  RETPARAMS="`gtkdialog3 --center --program=DEPS_DIALOG`"
-
-
+###krg---Start again --->>>
+. /usr/local/petget/check_deps.sh
 ###END###
